docs/guide/snaps-concepts.md

```diff
@@ -0,0 +1,147 @@
+# Concepts
+
+[[toc]]
+
+## Custom UI
+
+### Introduction
+
+Custom UI is a UI definition system used by various Snaps features. Rather than returning simply text strings, certain Snaps actions will have the ability to return a richer UI to express their output.
```

Comment:
The second sentence seems out of place as an introductory sentence. Maybe describe the scope and motivation rather than that certain snaps don't return text strings? "The Custom UI enables Snaps to [...]" or along similar lines.